src/gfx/renderer.cpp

```diff
@@ -0,0 +1,1010 @@
+      auto it = std::upper_bound(drawablesSorted.begin(), drawablesSorted.end(), sortableDrawable, comparison);
+      drawablesSorted.insert(it, sortableDrawable);
+    }
+
+    if (step.sortMode == SortMode::BackToFront) {
+      depthSortBackToFront(cachedPipeline, view);
+    }
+
+    // Creates draw call ranges based on DrawGroupKey
+    if (drawablesSorted.size() > 0) {
+      size_t groupStart = 0;
+      DrawGroupKey currentDrawGroupKey = drawablesSorted[0].key;
+
+      auto finishGroup = [&](size_t currentIndex) {
+        size_t groupLength = currentIndex - groupStart;
+        if (groupLength > 0) {
+          cachedPipeline.drawGroups.emplace_back(currentDrawGroupKey, groupStart, groupLength);
+        }
+      };
+      for (size_t i = 1; i < drawablesSorted.size(); i++) {
+        DrawGroupKey drawGroupKey = drawablesSorted[i].key;
+        if (drawGroupKey != currentDrawGroupKey) {
+          finishGroup(i);
+          groupStart = i;
+          currentDrawGroupKey = drawGroupKey;
+        }
+      }
+      finishGroup(drawablesSorted.size());
+    }
+  }
+
+  void resetPipelineCacheDrawables() {
+    for (auto &pair : pipelineCache) {
+      pair.second->resetDrawables();
+    }
+  }
+
+  WGPUBindGroup createTextureBindGroup(CachedPipeline &cachedPipeline, const TextureIds &textureIds) {
+    std::vector<WGPUBindGroupEntry> entries;
+    size_t bindingIndex = 0;
+    for (auto &id : textureIds.textures) {
+      entries.emplace_back();
+      entries.emplace_back();
+      WGPUBindGroupEntry &textureEntry = entries.data()[entries.size() - 2];
+      textureEntry.binding = bindingIndex++;
+      WGPUBindGroupEntry &samplerEntry = entries.data()[entries.size() - 1];
+      samplerEntry.binding = bindingIndex++;
+
+      TextureContextData *textureData = cachedPipeline.textureIdMap.resolve(id);
+      if (textureData) {
+        textureEntry.textureView = textureData->defaultView;
+        samplerEntry.sampler = textureData->sampler;
+      } else {
+        textureEntry.textureView = placeholderTexture->textureView;
+        samplerEntry.sampler = placeholderTexture->sampler;
+      }
+    }
+
+    WGPUBindGroupDescriptor textureBindGroupDesc{};
+    textureBindGroupDesc.layout = cachedPipeline.bindGroupLayouts[1];
+    textureBindGroupDesc.entries = entries.data();
+    textureBindGroupDesc.entryCount = entries.size();
+    return wgpuDeviceCreateBindGroup(context.wgpuDevice, &textureBindGroupDesc);
+  }
+
+  void renderDrawables(const DrawQueue &drawQueue, RenderDrawablesStep &step, ViewPtr view, Rect viewport, WGPUBuffer viewBuffer) {
+    WGPUDevice device = context.wgpuDevice;
+    WGPUCommandEncoderDescriptor desc = {};
+    WGPUCommandEncoder commandEncoder = wgpuDeviceCreateCommandEncoder(device, &desc);
+
+    WGPURenderPassDescriptor passDesc = {};
+    passDesc.colorAttachmentCount = 1;
+
+    WGPURenderPassColorAttachment mainAttach = {};
+    mainAttach.clearColor = clearColor;
+
+    // FIXME

```

Comment:
WHat to fix? Add more detail :)

Reply:
Yeah I'll turn it into a GitHub issue.

Reply:
Ah the issue was there, updated the comment to indicate what is missing